language-server/vscoqtop/lspManager.ml

```diff
@@ -257,9 +257,28 @@ let reset_observe_ids =
   Hashtbl.fold reset_doc_observe_id states
 
 let open_new_document uri text =
-  let vst, opts = get_init_state () in
+  let vst = get_init_state () in
+
+  let local_args =
+    let fname = DocumentUri.to_path uri in
+    let dir = Filename.dirname fname in
+    match CoqProject_file.find_project_file ~from:dir ~projfile_name:"_CoqProject" with
+    | None ->
+      log (Printf.sprintf "No project file found for %s" fname);
+      Coqargs.default
+    | Some f ->
+      let project = CoqProject_file.read_project_file ~warning_fn:(fun _ -> ()) f in
+      let args = CoqProject_file.coqtop_args_from_project project in
+      log (Printf.sprintf "Arguments from project file %s: %s" f (String.concat " " args));
+      fst @@ Coqargs.parse_args ~init:Coqargs.default args
+  in
+
+  let () = Vernacstate.unfreeze_full_state vst in
+  let () = Coqinit.init_document local_args in
```

Comment:
TODO optcomp
I guess for older coq we do nothing as init_runtime includes init_document

(we still get injections from the local _CoqProject so -noinit works but basically everything else is from the root coqproject)